AI #2071: Enhance AI usage guidelines (#2072)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: guidelines/contributors/development-processes.md

We prefer using merge commits, as they provide traceability and context on the features being merged. For feature branches (used for creating Pull Requests) a clean-up rebase before sharing for review is encouraged. Merge commits are absolutely preferred on the protected branches. Squash merges should be avoided due to the nature of how the original commits are not recorded in the commit graph of the target branch. For more information on how merges work in Git see: ([Different Merge Types in Git](https://lukemerrett.com/different-merge-types-in-git/))

## Content

### Authoring Examples

> When a contribution requires creating a series of examples, contributors must not draft the entire set all at once. Instead, authors are required to draft one or two representative examples and submit them to their respective Task Force for preliminary review.

Reviewer: It's a little unclear when we use BCP14 vs when we dont. Feel free to close and we can do a more formal pass on when to use terms in a future release, but making a suggestion just in case you agree. 

```suggestion
When a contribution requires creating a series of examples, contributors MUST NOT draft the entire set all at once. Instead, authors SHOULD draft one or two representative examples and submit them to their respective Task Force for preliminary review.
```

Author: Yeah... agree we should use BCP14.  And also, we don't use it at all in this document.  Recommend doing a wider pass in the future.  Closing with no action.
